filesystem/node.go

```diff
@@ -360,8 +384,33 @@ func (n *Node) Rename(ctx context.Context, req *fuse.RenameRequest, newDir fs.No
 	if !ok {
 		return fmt.Errorf("newDir is not a Node: %v", newDir)
 	}
-	if !n.Mode.IsDir() || !newNode.Mode.IsDir() {
+	if !n.isDir() || !newNode.isDir() {
 		return fuse.Errno(syscall.ENOTDIR)
 	}
 	return n.cfs.rename(n.ID, newNode.ID, req.OldName, req.NewName)
 }
+
+// Symlink creates a new symbolic link in the receiver node, which must
+// be a directory.
+func (n *Node) Symlink(_ context.Context, req *fuse.SymlinkRequest) (fs.Node, error) {
+	if !n.isDir() {
+		return nil, fuse.Errno(syscall.ENOTDIR)
+	}
+	// TODO(marc): have a max target length. If exceeded, either return ENAMETOOLONG,
+	// or store as slow link in a block.
```

Comment:
Returning `ENAMETOOLONG` makes sense to me. We could set this somewhat high (4KB). Storing the link in a separate block doesn't.